forms/models/elements/subform.js

```diff
@@ -256,7 +256,14 @@ define(function (require) {
 
       return new Promise(function (resolve, reject) {
         Promise.all(promises).then(function (values) {
-          resolve(values);
+          values = _.reduce(values, function (memo, value) {
+            if (value._action === 'remove' && (value.id === null || value.id === undefined)) {
+              return memo;
+            }
+            memo.push(value);
+            return memo;
+          }, []);
+          resolve(_.compact(values));
```

Comment:
i actually removed the compact, it was unnecessary and left over from a previous iteration where the forms model returned undefined. I decided to do it all in subforms rather than split the logic of removing over subform + form.